dipy/segment/featurespeed.pxd

```diff
@@ -0,0 +1,19 @@
+from cythonutils cimport Data2D, Shape
+
+
+cdef class Feature(object):
+    cdef int is_order_invariant
+
+    cdef Shape c_infer_shape(Feature self, Data2D streamline) nogil except *
```

Comment:
I agree the variable name should be something like `datum`. The reason I used Data2D instead of the fused type Data (that supports up to 7D) is that Cython's extension classes are not working with fused type! See http://trac.cython.org/ticket/833 .